src/main/kx/c.java

```diff
@@ -636,17 +637,21 @@ private void u() {
 
     public synchronized K.KBase k(K.KBase x, ProgressCallback progress) throws K4Exception, IOException {
         try {
-
+            cancelled = false;
```

Comment:
i also dont think this line is needed as default value is `false` already

Reply:
The way I understand it, the object can be reused for multiple queries. So if a new query is started, it should start in a non-cancelled state. I prefer to remove the initialization at the top.

Reply:
I dont see how it's re-used - every `QueryExecutor.execute` it creates a new `Worker`. but i may be missing something

Reply:
Well, maybe that's how it works out in practice, but we need to look at this code from a narrow perspective. This is the low level Java client API, which has no concept of query executors or workers. The API itself needs to be able to handle multiple queries on the same connection. Not initializing the variable before every query would make this misbehave.

Reply:
that's fair